text/0155-external-modules.md

```diff
@@ -0,0 +1,106 @@
+
+### Contrib components
+
+Any language SIG may have one or more “contrib” repos containing components contributed by external developers.
+This repository may have a separate set of approvers/maintainers than the core API/SDK repo.
+SIG maintainers are encouraged to have a simplified process to promote contributors to approver/maintainer role in this repository.
+Contrib repository may leverage the CODEOWNERS functionality of GitHub to assign maintainers to individual packages
+even if this means granting write permissions to the whole repo.
+The goal should be to distribute the load of reviewing PRs and accepting changes as much as possible.
+
+All components in contrib repository are expected to be included into Otel Registry following the usual process described above.
+
+It should be very easy for external contributors to include their components into contrib repo as opposed to hosting them separately.
+This way they can reuse existing infrastructure for testing, publishing, security scanning etc.
+Also this will greatly simplify responsibility transfer between different maintainers if their priorities change. It also promotes the development
+and maintenance of a single instrumentation package for each instrumentation source, so that work isn't spread amongst multiple parallel solutions.
+
+Language SIGs are encouraged to provide a testing harness to verify that component adheres to OpenTelemetry semantic conventions
+and [recommendations for Otel instrumentations design](https://docs.google.com/document/d/1YNRCg9fdjJgZRs56vvf7rfFPk06mhp781sWHYypOaAk/edit#)
+when OpenTelemetry starts publishing them.
+
+A high volume of contrib contributions presents a burden for language maintainers. There are two suggestions for tackling this:
+
+- Create an "experimental" folder within contrib. The contents of this folder are not reviewed or maintained by language repository maintainers, but
+many of the other benefits of being within a contrib repository remain.
+- Add more approvers and maintainers, perhaps some who exclusively focus on submissions to contrib.
+
+### External components
+
+If component authors for whatever reason want to host their contribution outside the Otel contrib repository they are free to do so (though we
+encourage all contributions to go into contrib or core language repositories). Their submission for inclusion into OpenTelemetry Registry is still
+welcomed, subject to the same process described above.
+
+### Distribution
+
+Whenever OpenTelemetry components are published to any repository other than OpenTelemetry Registry (such as npm registry or Maven Central)
+only core and contrib components can be published under "opentelemetry" namespace.
+Native and external components are to be published under their own namespace.
+
+In case Otel SIG provides any kind of "all-in-one" instrumentation distribution (e.g. Java and .NET do) they should include
+only core and contrib packages into it.
```

Comment:
Why this restriction? It's akin to saying that Jaeger binary (for example) may only include code from Jaeger repos, no 3rd party dependencies. If there's a high quality instrumentation hosted by 3rd party, I don't see why the distro cannot include it.

Reply:
+1.

Maybe be less restrictive.   

```
If Otel provides bundled instrumentation distributions ONE of the available options should only include instrumentation under the OpenTelemetry provenance.
```

For bundled instrumentation distributions WITH third party instrumentation, we need to make sure issues/support is directed to the appropriate location, but I think we can solve that issue.

Reply:
This restriction was included based on the feedback on the initial draft: https://docs.google.com/document/d/1YNRCg9fdjJgZRs56vvf7rfFPk06mhp781sWHYypOaAk/edit?disco=AAAAMSIJr4A

Reply:
The concern was about security risk.   I don't think this clause is the only way to deal with security concerns, nor does it alleviate security risk in any bundled offering of -contrib.   I think we need more discussion around this limtiation, I'd be very against making it without more thought on security overall.

Reply:
I tried to modify wording, PTAL again.